pkg/api/handlers/generic/containers.go

```diff
@@ -0,0 +1,306 @@
+		if err != nil {
+			handlers.InternalServerError(w, err)
+			return
+		}
+		switch state {
+		case define.ContainerStateStopped, define.ContainerStateExited:
+		default:
+			continue
+		}
+
+		deletedContainers = append(deletedContainers, ctnr.ID())
+		cSize, err := ctnr.RootFsSize()
+		if err != nil {
+			handlers.InternalServerError(w, err)
+			return
+		}
+		spaceReclaimed += uint64(cSize)
+
+		err = runtime.RemoveContainer(context.Background(), ctnr, false, false)
+		if err != nil && !(errors.Cause(err) == define.ErrCtrRemoved) {
+			handlers.InternalServerError(w, err)
+			return
+		}
+	}
+	report := types.ContainersPruneReport{
+		ContainersDeleted: deletedContainers,
+		SpaceReclaimed:    spaceReclaimed,
+	}
+	handlers.WriteResponse(w, http.StatusOK, report)
+}
+
+func LogsFromContainer(w http.ResponseWriter, r *http.Request) {
+	decoder := r.Context().Value("decoder").(*schema.Decoder)
+	runtime := r.Context().Value("runtime").(*libpod.Runtime)
+
+	query := struct {
+		Follow     bool   `schema:"follow"`
+		Stdout     bool   `schema:"stdout"`
+		Stderr     bool   `schema:"stderr"`
+		Since      string `schema:"since"`
+		Until      string `schema:"until"`
+		Timestamps bool   `schema:"timestamps"`
+		Tail       string `schema:"tail"`
+	}{
+		Tail: "all",
+	}
+	if err := decoder.Decode(&query, r.URL.Query()); err != nil {
+		handlers.Error(w, "Something went wrong.", http.StatusBadRequest, errors.Wrapf(err, "Failed to parse parameters for %s", r.URL.String()))
+		return
+	}
+
+	if !(query.Stdout || query.Stderr) {
+		msg := fmt.Sprintf("%s: you must choose at least one stream", http.StatusText(http.StatusBadRequest))
+		handlers.Error(w, msg, http.StatusBadRequest, errors.Errorf("%s for %s", msg, r.URL.String()))
+		return
+	}
+
+	name := mux.Vars(r)["name"]
+	ctnr, err := runtime.LookupContainer(name)
+	if err != nil {
+		handlers.ContainerNotFound(w, name, err)
+		return
+	}
+
+	var tail int64 = -1
+	if query.Tail != "all" {
+		tail, err = strconv.ParseInt(query.Tail, 0, 64)
+		if err != nil {
+			handlers.BadRequest(w, "tail", query.Tail, err)
+			return
+		}
+	}
+
+	var since time.Time
+	if _, found := mux.Vars(r)["since"]; found {
+		since, err = util.ParseInputTime(query.Since)
+		if err != nil {
+			handlers.BadRequest(w, "since", query.Since, err)
+			return
+		}
+	}
+
+	var until time.Time
+	if _, found := mux.Vars(r)["until"]; found {
+		since, err = util.ParseInputTime(query.Until)
+		if err != nil {
+			handlers.BadRequest(w, "until", query.Until, err)
+			return
+		}
+	}
+
+	options := &logs.LogOptions{
+		Details:    true,
+		Follow:     query.Follow,
+		Since:      since,
+		Tail:       tail,
+		Timestamps: query.Timestamps,
+	}
+
+	var wg sync.WaitGroup
+	options.WaitGroup = &wg
+
+	logChannel := make(chan *logs.LogLine, tail+1)
+	if err := runtime.Log([]*libpod.Container{ctnr}, options, logChannel); err != nil {
+		handlers.InternalServerError(w, errors.Wrapf(err, "Failed to obtain logs for Container '%s'", name))
```

Comment:
No capitalized errors.

Reply:
to be consistent with the other api, we must use caps.